portal-api/api-tests/v1/users/users.api-test.js

```diff
@@ -195,75 +196,150 @@ describe('a user', () => {
       };
 
       await createUser(newUser);
-      const { status, body } = await as(loggedInUser).get('/v1/users');
+      const { status, body } = await as(aNonAdmin).get('/v1/users');
 
       expect(status).toEqual(200);
-      expect(body.users[1].roles).toStrictEqual([READ_ONLY]);
+      expect(body.users.find((user) => user.username === 'TEMPORARY_USER').roles).toStrictEqual([READ_ONLY]);
     });
   });
 
   describe('PUT /v1/users', () => {
-    it('a user can be updated', async () => {
-      const response = await createUser(MOCK_USER);
-      const createdUser = response.body.user;
+    describe('as admin', () => {
+      it('a user\'s details can be updated', async () => {
+        const response = await createUser(MOCK_USER);
+        const createdUser = response.body.user;
+
+        const updatedUserCredentials = {
+          roles: [CHECKER, MAKER],
+          firstname: 'NEW_FIRSTNAME',
+          surname: 'NEW_SURNAME',
+          'user-status': STATUS.BLOCKED
+        };
+
+        const { status } = await as(anAdmin).put(updatedUserCredentials).to(`/v1/users/${createdUser._id}`);
+
+        expect(status).toEqual(200);
+
+        const { body } = await as(anAdmin).get(`/v1/users/${createdUser._id}`);
+
+        expect(body).toEqual(expect.objectContaining({
+          roles: [CHECKER, MAKER],
+          firstname: 'NEW_FIRSTNAME',
+          surname: 'NEW_SURNAME',
+          'user-status': STATUS.BLOCKED,
+        }));
+      });
 
-      const updatedUserCredentials = {
-        roles: [CHECKER, MAKER],
-      };
+      it('a user\'s password can be updated', async () => {
+        const response = await createUser(MOCK_USER);
+        const createdUser = response.body.user;
 
-      await as(loggedInUser).put(updatedUserCredentials).to(`/v1/users/${createdUser._id}`);
+        const updatedUserCredentials = {
+          password: 'AbC1234!',
+          passwordConfirm: 'AbC1234!'
+        };
 
-      const { status, body } = await as(loggedInUser).get(`/v1/users/${createdUser._id}`);
+        const { status } = await as(anAdmin).put(updatedUserCredentials).to(`/v1/users/${createdUser._id}`);
```

Comment:
This doesn't verify that the password has been updated, should I add something like
``` 
const { status } = await as().post({ username, password }).to('/v1/login');
expect(status).toEqual(200);
```

Reply:
I think performing that test is slightly too granular for an api test as it should only be possible to get a 200 status code if a user has successfully been updated, so you can make the assumption that a user was updated if a 200 is received - the actual update functioning is something I'd expect to check in an e2e test. Not 100% sure what the best practice is though